tools/tsdb-series/main.go

```diff
@@ -157,38 +226,53 @@ func formatTime(ts time.Time) string {
 	return ts.UTC().Format(time.RFC3339Nano)
 }
 
-func computeChunkStats(chr tsdb.ChunkReader, chks []chunks.Meta) (time.Time, time.Time, int) {
+func computeChunkStats(chr tsdb.ChunkReader, chks []chunks.Meta, minTimeFilter, maxTimeFilter int64) (minTime int64, maxTime int64, totalSamples int) {
 	if len(chks) == 0 {
-		return time.Time{}, time.Time{}, 0
+		return 0, 0, 0
 	}
 
-	minTS := chks[0].MinTime
-	maxTS := chks[len(chks)-1].MaxTime
-	totalSamples := 0
-
 	for _, cm := range chks {
-		c, it, err := chr.ChunkOrIterable(cm)
+		c, itb, err := chr.ChunkOrIterable(cm)
 		if err != nil {
 			level.Error(logger).Log("msg", "failed to open chunk", "err", err, "chunk", cm.Ref)
-			return time.Time{}, time.Time{}, 0
+			return 0, 0, 0
 		}
 
+		var it chunkenc.Iterator
 		if c != nil {
-			totalSamples += c.NumSamples()
-		} else if it != nil {
-			i := it.Iterator(nil)
-			for i.Next() != chunkenc.ValNone {
-				totalSamples++
-			}
-			if err := i.Err(); err != nil {
-				level.Error(logger).Log("msg", "got error while iterating chunk", "chunk", cm.Ref, "err", err)
-				return time.Time{}, time.Time{}, 0
-			}
+			it = c.Iterator(nil)
+		} else if itb != nil {
+			it = itb.Iterator(nil)
 		} else {
 			level.Error(logger).Log("msg", "can't determine samples in chunk", "chunk", cm.Ref)
-			return time.Time{}, time.Time{}, 0
+			return 0, 0, 0
+		}
+
+		for it.Next() != chunkenc.ValNone {
+			include := true
+			ts := it.AtT()
+
+			if minTimeFilter > 0 && ts < minTimeFilter {
+				include = false
+			}
+			if maxTimeFilter > 0 && ts > maxTimeFilter {
+				include = false
+			}
+			if include {
+				totalSamples++
+				if minTime == 0 {
+					minTime = ts
+				}
```

Comment:
Is it a given that we're iterating in increasing time order?

Reply:
TSDB library produces chunks with samples with increasing timestamps, so it *should* be.